src/trace-plugin-loader.js

```diff
@@ -116,16 +116,20 @@ function activate(agent) {
       if (!patchSet) {
         // Load the plugin object
         var plugin = originalModuleLoad(instrumentation.file, module, false);
-        patchSet = [];
+        patchSet = {};
```

Comment:
It seems like `intrumentation.patches[moduleRoot]` is treated as an object everywhere else.